### DISCUSSION TITLE
If a built next app is all Static, can I deploy it to a static hosting? and how?

### DISCUSSION THREAD

**Discussion by iam-yan** [vercel/next.js#13108] [Help]
Without any usage of getInitialProps, getServerSideProps and getStaticProps, the app is all Static. In this case can I deploy it to a static hosting? And how?

**Comment by rafaelalmeidatk** [comment]
You can generate static HTML with `next export` that can be utilized in any static hosting service: https://nextjs.org/docs/advanced-features/static-html-export

**Reply by iam-yan** [reply]
@rafaelalmeidatk Thanks. But I have dynamic route and the page would fetch data from api based on the path. According to this discussion https://github.com/zeit/next.js/discussions/12121#discussioncomment-6266
`next export` sounds like not a good fit?

**Reply by iam-yan** [reply]
I also use catch all route. In this case what is the best practice to build and deploy the app into a static hosting?

btw... I have tried to export and deploy it to firebase hosting. But I could not access to pages other then index.html, even paths like `/project/new.html`... What might I have missed?

**Reply by w3bdesign** [reply]
@juuyan Have you tried setting up custom routes with vercel.json?

Take a look here:

https://vercel.com/docs/configuration#routes

**Reply by rafaelalmeidatk** [reply]
`next export` is the only way to generate static HTML files, all other methods will require a Node server. I has been a while since I built a static HTML app but you need to tell your hosting service to route the dynamic URL to a HTML file, like `/user/foo` to the `/user/[id].html` file

**Reply by rafaelalmeidatk** [reply]
> I also use catch all route. In this case what is the best practice to build and deploy the app into a static hosting?

I tested it but looks like the catch all routes aren't supported by `next export` 🤔
What you would have to do is route all the paths to a specific file, then render the page based on `location.pathname`. Keep in mind this will not benefit from code-splitting of individual pages since you are essentially bundling all the code from these pages into a single entrypoint

**Reply by iam-yan** [reply]
> `next export` is the only way to generate static HTML files, all other methods will require a Node server. I has been a while since I built a static HTML app but you need to tell your hosting service to route the dynamic URL to a HTML file, like `/user/foo` to the `/user/[id].html` file

@rafaelalmeidatk  I am bit confused. According to the document, the 'Static Export' is not necessary and could be replaced with regular build thanks to Automatic Static Optimization, isn't it? If so, why other methods would always require a server?
>If your pages don't have getInitialProps you may not need next export at all; next build is already enough thanks to Automatic Static Optimization.

**Reply by iam-yan** [reply]
@w3bdesign Thanks. But I was trying to use firebase hosting. Do you have any idea on how to make it work?
